documentation/cobigen-cli_debugging.asciidoc

```diff
@@ -0,0 +1,38 @@
+= Debugging Cobigen-CLI:
+
+1. Open Debug Configurations by clicking on "Run > Debug Configurations" in Eclipse.
+
+2. Click on "Java Application" and then click on "New".
+
+3. Select the "cli" project and set the Main class to "com.devonfw.cobigen.cli.CobiGenCLI".
+
+4. Add any arguments that you would normally use when running the CLI.
+
+5. Make sure that every CLI command is possible. For example, you can use the command "g 'PATH TO EmployeeEntity.java'".
+
+6. Under the File System tab, set the Working Directory to the directory where the CLI is opened.
+
+7. Open cobigen.bat and add the following flag +
+*-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=1044* +
+to the command before "-cp": +
+image:images/debugging/cobigen-bat.png[]
+
+8. Go to the Remote Java Application tab in Debug Configurations and click on "New".
```

Comment:
Resolved

Reply:
This might be a misunderstanding.
I meant that if you want to debug tests, you need to adjust the cobigen.bat.
The debug setup which was done above already works fine but uses the real CobiGen home directory instead. That's why we need to warn the user about this behavior.
A good idea would be to separate the steps for the "simple" debugging and the test debugging.

You could also add a table of contents at the beginning of the document too:
```
:toc:
toc::[]
```